be/src/exec/pipeline/scan/olap_scan_context.h

```diff
@@ -40,14 +42,52 @@ using OlapScanContextPtr = std::shared_ptr<OlapScanContext>;
 class OlapScanContextFactory;
 using OlapScanContextFactoryPtr = std::shared_ptr<OlapScanContextFactory>;
 
+class JitRewriterConcurrent {
+public:
+    JitRewriterConcurrent(int32_t dop) : _dop(dop), _barrier(), _errors(0), _id(0) {}
+    Status rewrite(std::vector<ExprContext*>& expr_ctxs, ObjectPool* pool, bool enable_jit);
+
+private:
+    // TODO: use c++20 barrier after upgrading gcc
+    class Barrier {
```

Comment:
std::call_once is enough

Reply:
not really, if there are more than one expr, we can concurrently jit them, as jit may take tens of ms time.